openpype/lib/attribute_definitions.py

```diff
@@ -433,38 +434,93 @@ def __init__(self, key, items, default=None, **kwargs):
                 " defined values on initialization."
             ).format(self.__class__.__name__))
 
-        items = collections.OrderedDict(items)
-        if default not in items:
-            for _key in items.keys():
-                default = _key
+        items = self.prepare_enum_items(items)
+        item_values = [item["value"] for item in items]
+        if default not in item_values:
+            for value in item_values:
+                default = value
                 break
 
         super(EnumDef, self).__init__(key, default=default, **kwargs)
 
         self.items = items
+        self._item_values = set(item_values)
 
     def __eq__(self, other):
         if not super(EnumDef, self).__eq__(other):
             return False
 
-        if set(self.items.keys()) != set(other.items.keys()):
-            return False
-
-        for key, label in self.items.items():
-            if other.items[key] != label:
-                return False
-        return True
+        return self.items == other.items
 
     def convert_value(self, value):
-        if value in self.items:
+        if value in self._item_values:
             return value
         return self.default
 
     def serialize(self):
         data = super(TextDef, self).serialize()
-        data["items"] = list(self.items)
+        data["items"] = copy.deepcopy(self.items)
         return data
 
+    @staticmethod
+    def prepare_enum_items(items):
+        """Convert items to unified structure.
+
+        Output is a list where each item is dictionary with 'value'
+        and 'label'.
+
+        ```python
+        # Example output
+        [
+            {"label": "Option 1", "value": 1},
+            {"label": "Option 2", "value": 2},
+            {"label": "Option 3", "value": 3}
+        ]
+        ```
+
+        Args:
+            items (Union[Dict[str, Any], List[Any], List[Dict[str, Any]]): The
+                items to convert.
+
+        Returns:
+            List[Dict[str, Any]]: Unified structure of items.
+        """
+
+        output = []
+        if isinstance(items, dict):
+            for value, label in items.items():
+                output.append({"label": label, "value": value})
+
+        elif isinstance(items, (tuple, list, set)):
+            for item in items:
+                if isinstance(item, dict):
+                    # Test if value is available
+                    item["value"]
```

Comment:
What is this, this looks weird.

Reply:
It would raise `KeyError` if `"value"` is missing.

Reply:
Yeah, I would prefer explicit.